indico/modules/events/registration/controllers/management/reglists.py

```diff
@@ -335,20 +346,73 @@ def _process(self):
 
 
 class RHRegistrationsPrintBadges(RHRegistrationsActionBase):
+
+    normalize_url_spec = {
+        'locators': {
+            lambda self: self.regform
+        },
+        'preserved_args': {'uuid', 'template_id'}
+    }
+
+    def _checkParams(self, params):
+        RHRegistrationsActionBase._checkParams(self, params)
+        self.template = DesignerTemplate.get_one(request.view_args['template_id'])
+
+    def _checkProtection(self):
+        RHRegistrationsActionBase._checkProtection(self)
+
+        # Check that template belongs to this event or a category that
+        # is a parent
+        if self.template.owner != self.event_new and self.template.owner.id not in self.event_new.category_chain:
+            raise Forbidden
+
+    def _process(self):
+        self.commit = False
```

Comment:
probably not needed since you only read from the cache and don't modify anything (I'd only set it in places where may actually have some changes we don't want to commit)